### DISCUSSION TITLE
How to save extension data and access it after loading the project?

### DISCUSSION THREAD

**Discussion by KhandamovA** [TurboWarp/extensions#794] [Q&A] [answered]
How to save extension data and access it after loading the project?

**Comment by LilyMakesThings** [comment, marked as answer]
You can't lmao, been wanting to do that for a while now

**Reply by KhandamovA** [reply]
sadly, thanks for your reply.

**Reply by yuri-kiss** [reply]
so localstorage wont work?

**Reply by LilyMakesThings** [reply]
Not for what they want to do, no

**Reply by yuri-kiss** [reply]
what do they want to do?
explain pls.

**Reply by LilyMakesThings** [reply]
save extension data persistently, not project data. It's something on garbo's radar because of my discord extension needing it.

**Reply by yuri-kiss** [reply]
wdym by persistently, localstorage saves across sessions and can be accessed by extensions

**Reply by LilyMakesThings** [reply]
read it again

**Reply by yuri-kiss** [reply]
oh save the data in the sb3?
or `How to save extension data and access it after loading the project?` by data do they mean like a config, and you can load it after with localstorage otherwise the sb3 is my only other answer

**Reply by yuri-kiss** [reply]
the extension could always just write to a variable.
and store it there

**Reply by yuri-kiss** [reply]
```javascript
vm._toJSON = vm.toJSON;
vm.toJSON = function(optTargetId, serializationOptions) {
  let json = vm._toJSON(optTargetId, serializationOptions);
  json = JSON.parse(json);
  json.targets[0].variables["iy]C)2yZ=OVowUF}kf}_"] = ["", "test"]; //save the data here.
  json = JSON.stringify(json);
  return json;
}
```

**Reply by KhandamovA** [reply]
to store all the necessary extension data, I already chose list

**Comment by GarboMuffin** [comment]
Soon

**Reply by Skyhigh173** [reply]
soon™️
